### ISSUE TITLE
Reproduce

### ISSUE THREAD

**Issue by ShaobinChen-AH** [SAP-samples/acl2022-self-contrastive-decorrelation#2]
Hi! I notice that some hyper-parameters in your pre-trained model (see config.json) are slightly different from that in your paper.
For example, the hyperparameter task-beta is 1.05 and 0.005 in ReadMe and paper respectively. Which value should I use when I reproduce your results? 
I also want to know the value of temp, which you did not mention in your paper.
Thanks a lot.

**Comment by TJKlein** [comment]
Hi,

Yes, that's true. In the first version of the code a .sum() was used instead of a .mean() for self-contrast:

https://github.com/SAP-samples/acl2022-self-contrastive-decorrelation/blob/b990e4f8ff803473ba58eac8c67ef35acc1fd0ce/scd/models.py#L272


In order to unify the code and make the loss independent of batch-size later the mean was adopted. 

With mean(.) we got some slightly different hyperparameters with comparable performance, though.

So to replicate results identically, use .sum() and choose use 1.05. 

Assuming a batch-size of 192, 1.05 / 192 w should give some value around 0.0054*****.

The temp value is unchanged to SimCSE. I believe it is 0.05.


I hope that helps.

**Comment by ShaobinChen-AH** [comment]
Excuse me. I do not understand explicitly what you said. Do you mean that I need to change the code .mean() to .sum() in Line 272 and use the value of 1.05? 


| |
chenshaobin000001
|
|
***@***.***
|
签名由网易邮箱大师定制
On 4/20/2022 19:22，Tassilo ***@***.***> wrote：

Hi,

Yes, that's true. In the first version of the code a .sum() was used instead of a .mean() for self-contrast:

https://github.com/SAP-samples/acl2022-self-contrastive-decorrelation/blob/b990e4f8ff803473ba58eac8c67ef35acc1fd0ce/scd/models.py#L272

In order to unify the code and make the loss independent of batch-size later the mean was adopted.

With mean(.) we got some slightly different hyperparameters with comparable performance, though.

So to replicate results identically, use .sum() and choose use 1.05.

Assuming a batch-size of 192, 1.05 / 192 w should give some value around 0.054*****.

The temp value is unchanged to SimCSE. I believe it is 0.05.

I hope that helps.
